pkg/endpointslice/map.go

```diff
@@ -98,7 +98,11 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 		return
 	}
 
-	cluster, ok := es.Labels[constants.LabelSourceCluster]
+	cluster, ok := es.Labels[constants.MCSLabelSourceCluster]
+
+	if !ok {
+		cluster, ok = es.Labels[constants.LighthouseLabelSourceCluster]
```

Comment:
Perhaps we don't need to check the legacy label here since the endpoints controller should automatically update each object with the new labels on startup.

Reply:
It should, but this map is populated in coredns at startup too. So for upgrade scenarios [old object without mcs label] it may not have the label yet. It will still work, but there will be a small window where we may refuse requests even when available. Am okay either ways.

Reply:
I’d rather keep the lenient checks; I’ve added comments to indicate when they can be removed.